tests/units/TicketTest.php

```diff
@@ -294,4 +295,406 @@ public function testEscalationWithMandatoryFields()
+        $this->assertGreaterThan(0, $group1_id);
+
+        $user_group1 = new \Group_User();
+        $user_group1->add([
+            'users_id' => $user1->getID(),
+            'groups_id' => $group1->getID()
+        ]);
+        $this->assertGreaterThan(0, $user_group1->getID());
+
+        // Create ticket
+        $ticket = new \Ticket();
+        $t_id = $ticket->add([
+            'name' => 'Escalation Test',
+            'content' => '',
+            '_actors' => [
+                'assign' => [
+                    [
+                        'items_id' => $group1->getID(),
+                        'itemtype' => 'Group'
+                    ],
+                ],
+            ]
+        ]);
+
+        // Check group linked to the ticket
+        $ticket_group = new \Group_Ticket();
+        $this->assertEquals(1, count($ticket_group->find(['tickets_id' => $t_id])));
+
+        $ticket_user = new \Ticket_User();
+        $this->assertEquals(0, count($ticket_user->find(['tickets_id' => $t_id])));
+
+        $this->assertTrue($ticket->update(
+            [
+                'id' => $t_id,
+                '_actors' => [
+                    'assign' => [
+                        [
+                            'items_id' => $user1->getID(),
+                            'itemtype' => 'User'
+                        ],
+                        [
+                            'items_id' => $group1->getID(),
+                            'itemtype' => 'Group'
+                        ],
+                    ]
+                ]
+            ]
+        ));
+
+        // Check group linked to the ticket
+        $ticket_group = new \Group_Ticket();
+        $this->assertEquals(1, count($ticket_group->find(['tickets_id' => $t_id])));
+
+        $ticket_user = new \Ticket_User();
+        $this->assertEquals(1, count($ticket_user->find(['tickets_id' => $t_id])));
+
+        $itil_cat = new \ITILCategory();
+        $itil_cat_id = $itil_cat->add([
+            'name' => 'Cat1',
+        ]);
+
+        $this->assertTrue($ticket->update(
+            [
+                'id' => $t_id,
+                'itilcategories_id' => $itil_cat_id,
+                '_actors' => [
+                    'assign' => [
+                        [
+                            'items_id' => $user1->getID(),
+                            'itemtype' => 'User'
+                        ],
+                        [
+                            'items_id' => $group1->getID(),
+                            'itemtype' => 'Group'
+                        ],
+                    ]
+                ]
+            ]
+        ));
+
+        $ticket_group = new \Group_Ticket();
+        $this->assertEquals(1, count($ticket_group->find(['tickets_id' => $t_id])));
+
+        $ticket_user = new \Ticket_User();
+        $this->assertEquals(1, count($ticket_user->find(['tickets_id' => $t_id])));
+    }
+
+    private function testAssignGroupToTicketWithCategoryProvider()
```

Comment:
We need to cover all possible cases, and as there are 4 parameters, that makes 16 possible combinations.

```
0 0 0 0
0 0 0 1
0 0 1 0
0 0 1 1 missing
0 1 0 0 missing
0 1 0 1
0 1 1 0 missing
0 1 1 1
1 0 0 0
1 0 0 1
1 0 1 0
1 0 1 1 missing
1 1 0 0 missing
1 1 0 1
1 1 1 0
1 1 1 1
```